addOns/pscanrulesBeta/src/test/java/org/zaproxy/zap/extension/pscanrulesBeta/InformationDisclosureInURLUnitTest.java

```diff
@@ -0,0 +1,222 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesBeta;
+
+import static org.junit.Assert.assertEquals;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+public class InformationDisclosureInURLUnitTest
+        extends PassiveScannerTest<InformationDisclosureInURL> {
+
+    private static final String URI = "http://example.com/";
+    private static final String BODY = "Some text in the response, doesn't matter.\nLine 2\n";
+    private HttpMessage msg;
+
+    @Override
+    protected InformationDisclosureInURL createScanner() {
+        return new InformationDisclosureInURL();
+    }
+
+    @Override
+    public void setUpZap() throws Exception {
+        super.setUpZap();
+
+        Path xmlDir =
+                Files.createDirectories(
+                        Paths.get(
+                                Constant.getZapHome(),
+                                InformationDisclosureInURL.URLSensitiveInformationDir));
+        Path testFile = xmlDir.resolve(InformationDisclosureInURL.URLSensitiveInformationFile);
+        Files.write(testFile, Arrays.asList(" user", "password", "# notused", "session "));
+    }
+
+    protected HttpMessage createHttpMessageWithRespBody(String responseBody)
+            throws HttpMalformedHeaderException, URIException {
+
+        HttpRequestHeader requestHeader = new HttpRequestHeader();
+        requestHeader.setURI(new URI(URI, false));
+
+        msg = new HttpMessage();
+        msg.setRequestHeader(requestHeader);
+        msg.setResponseBody(responseBody);
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n"
+                        + "Server: Apache-Coyote/1.1\r\n"
+                        + "Content-Type: text/plain\r\n"
+                        + "Content-Length: "
+                        + responseBody.length()
+                        + "\r\n");
+        return msg;
+    }
+
+    @Test
+    public void sensitiveInfoInURLParamName() throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String testURI = URI + "foo?bar=whodat&UserName=Jonathon&what=up";
+        HttpMessage msg = createHttpMessageWithRespBody(BODY);
```

Comment:
Every test method seems to pass `BODY`, why not just build the message that way to start with?

(9 occurrences of `@Test`, 9 occurrences of `createHttpMessageWithRespBody(BODY)`)

Reply:
Might be worth passing and setting `testURI` instead of doing it in each test method.